### PR TITLE
update DG for Login feature

### PR DESCRIPTION
**Pull request by Acerizm**
1. Added more use case descriptions and activity diagrams for login feature
2. Added more user stories for filter command and login feature
3. Added 2 extra appendix for `Planned Enhancements` and `Limitations`
4. Added a custom rake based on https://github.com/nus-cs2103-AY1920S2/forum/issues/105 for login activity diagram as PlantUML has no support for rakes

### CODE REVIEW

File: docs/images/Login.png
Comment (joellow88) [inline docs/images/Login.png:1]: The activity diagram here seems slightly confusing. If I understand correctly, the login screen is the screen that prompts you for a password, that only appears if the user enters ModCheck more than once. While ModCheck will prompt the user to create a password only if it is the first time the user enters ModCheck.

However, as the join node is before the "ModCheck requests Student to create a password" node here, the user will be requested to create a password every time he logs in (regardless of whether it is his first time logging in or not). I'm not sure if that is your intention in this activity diagram.

File: docs/images/CreatePassword.png
Comment (joellow88) [inline docs/images/CreatePassword.png:1]: Might be better to use a while loop here (after the node "ModCheck displays error message", going back to the branch node), but the meaning is understandable in this diagram as well.

Comment (Acerizm) [reply docs/images/CreatePassword.png:1]: I tried to use a while loop but it's very difficult to do using plantUML while following CS2103 convention. I resorted to repeating the same action. @joellow88

File: docs/images/Login.png
Comment (Acerizm) [reply docs/images/Login.png:1]: Oops thanks for pointing that out. Will rectify it! @joellow88

Comment (Acerizm) [reply docs/images/Login.png:1]: @joellow88 is the login activity diagram better in the latest commit?

Comment (joellow88) [reply docs/images/Login.png:1]: Looks good, thanks for rectifying it!

File: docs/images/CreatePassword.png
Comment (joellow88) [reply docs/images/CreatePassword.png:1]: I see, this diagram should be fine then!

**Review by joellow88 — APPROVED** [review]

**Pull request merged**